src/client/cached.go

```diff
@@ -75,6 +75,12 @@ func (d *DiscoveryCacher) discoverAndCache(timeout time.Duration) (HTTPClient, e
 	return client, nil
 }
 
+func expired(creationTimestamp int64, ttl time.Duration) bool {
+	expirationTime := time.Unix(creationTimestamp, 0).Add(ttl)
+
+	return time.Now().After(expirationTime)
+}
```

Comment:
I think `return time.Since(time.Unix(creationTimestamp, 0)) > ttl` is correct then: cache **expired** when time **since** **creation** is bigger than **TTL**.